lib/adapters/xhr.js

```diff
@@ -118,8 +118,12 @@ module.exports = function xhrAdapter(config) {
     // Add headers to the request
     if ('setRequestHeader' in request) {
       utils.forEach(requestHeaders, function setRequestHeader(val, key) {
-        if (typeof requestData === 'undefined' && key.toLowerCase() === 'content-type') {
-          // Remove Content-Type if data is undefined
+        if (key.toLowerCase() === 'content-type' &&
+          ((typeof requestData === 'undefined' && config.method.toLowerCase() === 'post') ||
+            (typeof config.params === 'undefined' && config.method.toLowerCase() === 'get'))
```

Comment:
do you mind using the utils' `isUndefined` for these 2?

Reply:
I would be happy to do so. Thank you for your advice.